### DISCUSSION TITLE
Lighthouse V2 : Position Instability and Signal Loss

### DISCUSSION THREAD

**Discussion by aalessi3** [bitcraze/discussions#468] [Q&A]
Hello, I have been struggling to get accurate, stable or repeatable state estimation using the Lighthouse positioning system. Any help would be greatly appreciated.

I am currently working with a single CF and a pair of Lighthouse V2 base stations. My flight area is approximately 3 x 3 x 2.5 (meters). My base stations are positioned 2.5m high and positioned such that their beams are perpendicular to each other (i.e at vertices A and B of a square). There are no reflective surfaces in my flight area, simply  blank walls and a net. I am facing the following issues.

1. Signal reception or loss for a given state is not repeatable. For a given position and orientation the cfclient will indicate that the CF is receiving signal from both base stations. Once moved and placed back in that same exact position and orientation, the client will indicate the CF is only receiving signal from 1 or neither base station. 

2. Once I am able to complete the geometry estimation process my ability to maintain a stable holding position varies. Upon completing geometry estimation I observe the estimated global position of the base stations and the values seem reasonable. I then instruct my CF to hover a set distance (.5m) off the ground and hold its position. In some instances this is successful, in some instances there is a large amount of drift in all directions (largest in z) and in some instances this drift is so large the cf seems to loss either base station and crashes.

These issues prevail despite trying multiple different CF's, different LH decks, different base station configurations, a completely different flight location and both "cross beam" and "sweep angle" lighthouse.method params.

**Comment by krichardsson** [comment]
Hi,
this sounds problematic! I have not really seen this type of problem before but let's dig into it and try to figure out what it might be.
Thanks for a very good description of the problems and a thorough, structured investigation!

Your system setup looks very good to me, and there is nothing there that I would suspect could cause problems. The placement of the base stations is good, the size of the space is reasonable and you mention that there are no reflections. All good!
 
When I was reading the post, my initial though was that there might be some issue on the Crazyflie side, for instance bad connection to the deck, but since you have tried with multiple Crazyflies and decks we can rule this out as well.
Perhaps there is some sort of problem with the base station(s)?

If I understand correctly, you loose data from the base stations and it is so bad that you even have a hard time to get through the geometry estimation. I have no idea what the root cause might be, but I will start with some random questions.
1. What firmware are you using in the Crazyflie?
2. if you log the angles from the base stations, does it look OK? Use the ["Plotter" tab](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/plotter_tab/) in the client and log the `lighthouse.angle0x`, `lighthouse.angle0y`, `lighthouse.angle1x` and `lighthouse.angle1y` variables. These variables should show the horizontal/vertical angles to the two base stations, where the base stations are the reference frame. As long as you receive a base station the angle should be updated and the curve should be smooth.
3. Have you tried to turn off one of the base stations? Can you get a stable reception in this case? 
4. I assume the base stations are set to channels 1 and 2?

Some random comments:
There is a [bug in the FPGA code](https://github.com/bitcraze/lighthouse-fpga/issues/14) that prevents the deck from decoding the signal from the base stations at some angles, mainly when the CF is pointing towards a base station (or 90 degrees to it). I do not think this is the main problem but it might something you see as well that adds to the confusion. If this is the case you should get reception again by rotating the CF a few degrees.

The deck must receive the light sweeps from a base station by 2 or more sensors on the deck to be able to decode the signal. Is there any reason that this might not be the case, for instance prop guards, other hardware that is blocking the view or similar? My guess is that you are  using a clean CF with the lighthouse deck mounted on top and nothing else, if this is true all should be good.
